### DISCUSSION TITLE
How to build the developer build of the documentation?

### DISCUSSION THREAD

**Discussion by metab0t** [coin-or/CppAD#193] [Development]
As mentioned in https://github.com/coin-or/CppAD/issues/192#issuecomment-1973084796, the developer build of the documentation contains examples and documentation for the val_graph representation

**Comment by bradbell** [comment]
You will need python and will have to install xrst
https://xrst.readthedocs.io/latest/

All of the file names below are relative to the CppAD Distributon directory
https://cppad.readthedocs.io/latest/download.html#distribution-directory

 If you have a bash shell, you can build the user documentation with the following command
`bin/run_xrst.sh -dev`
and the developer documentation with
`bin/run_xrst.sh +dev`
This will include any changes you make to the documentation. Note that this version uses the sphinx_rtd_theme and the web version uses the furo theme to build the documentation (the sphinx_rtd_theme is faster and so is used for testing).

You can then open the top level page of the documentation, for example using firefox with the command
`firefox build/html/index.html`

If you specify `+dev` , there will be an extra link in the top level left hand frame called `devel` . All of the developer documentation is below this. The developer documentation used to be in doxygen, and some has not yet been converted to xrst.

**Comment by bradbell** [comment]
There was a recent fix to the xrst version of the search; see the heading 02-03 on
https://xrst.readthedocs.io/latest/2024.html#mm-dd-03-02

Note that this does not affect the stable release xrst-2024.0.0 on pypi, just the test versions on 
https://test.pypi.org/project/xrst/
